utils/logger.go

```diff
@@ -17,12 +21,41 @@ const (
 
 // NewLogger creates and returns a new instance of the Logger type. This
 // initializes the logger with a tagged logrus Entry initialized with the
-// version string and hostname.
-func NewLogger(version string) Logger {
+// version string, hostname and two logging options (syslog & logpath)
+// By default logrus write to STDERR, NewLogger use STDOUT by default
+// If syslog is true the logger will use local syslog method and STDOUT
+// with basic text format.
+// If syslog is false 'logpath' will be evaluated as log file path with rotation
+// and also will be used STDOUT, both with json format.
+func NewLogger(version string, syslogEnabled bool, logpath string) Logger {
 	log := logrus.New()
+	log.Out = os.Stdout
+	if syslogEnabled {
+		hook, err := logrus_syslog.NewSyslogHook("", "", syslog.LOG_INFO, "device-hub")
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		log.Formatter = &logrus.TextFormatter{
+			DisableColors: true,
+		}
+		log.Hooks.Add(hook)
+		logger := log.WithFields(defaultFields(version))
+		return &l{entry: logger}
+	}
+
 	log.Formatter = new(logrus.JSONFormatter)
 	log.Level = logrus.InfoLevel
-	log.Out = os.Stdout
+	// TODO Parameterize Maximums
+	if len(logpath) != 0 {
```

Comment:
I agree this is very ugly, with the changes suggested on main.go this code should check against a bool flag and we will have a default log file name: device-hub.log.